src/models/Project/index.js

```diff
@@ -63,6 +66,70 @@ class Project extends Model {
     return this.ownerId === ownerId
   }
 
+  async stats() {
+    const translationKeysCount = await TranslationKey.query()
+      .join('projects as p', 'translationKeys.projectId', 'p.id')
+      .where('p.id', '=', this.id)
+      .count()
+      .pluck('count')
+      .first()
+
+    const localesCount = await Locale.query()
+      .join('projects_locales as r', 'locales.id', 'r.localeId')
+      .join('projects as p', 'r.projectId', 'p.id')
+      .where('p.id', '=', this.id)
+      .count()
+      .pluck('count')
+      .first()
+
+    if (translationKeysCount === '0') {
+      return new Stats({
+        missingTranslationsCount: 0,
+        newKeysCount: 0,
+        completePercentage: 100,
+        translationKeysCount: 0,
+        localesCount,
+        projectsCount: null
+      })
+    }
+
+    const expectedTranslationValuesCount = translationKeysCount * localesCount
+    const actualTranslationValuesCount = await TranslationValue.query()
+      .join(
+        'translationKeys as tk',
+        'translationValues.translationKeyId',
+        'tk.id'
+      )
+      .join('projects as p', 'tk.projectId', 'p.id')
+      .where('p.id', '=', this.id)
+      .count()
+      .pluck('count')
+      .first()
+
+    const missingTranslationsCount =
+      expectedTranslationValuesCount - actualTranslationValuesCount
+
+    const completePercentage = Math.round(
+      (actualTranslationValuesCount / expectedTranslationValuesCount) * 100
+    )
+
+    const newKeysCount = await TranslationKey.query()
+      .join('projects as p', 'translationKeys.projectId', 'p.id')
+      .where('translationKeys.isNew', '=', true)
+      .count()
+      .pluck('count')
+      .first()
+
+    return new Stats({
+      missingTranslationsCount,
+      newKeysCount,
+      completePercentage,
+      translationKeysCount,
+      localesCount,
+      projectsCount: null
```

Comment:
why always setting it to null?

Reply:
Because those are project stats (so not interested in getting that particular property at this point)